meta-mbl-distro/scripts/create-update-payload

```diff
@@ -0,0 +1,191 @@
+    parser.add_argument("-b", "--bootloader-slots", nargs="+")
+    parser.add_argument(
+        "-o",
+        "--tar-output-path",
+        type=lambda x: pathlib.Path(x).absolute(),
+        default="payload.tar",
+    )
+    parser.add_argument(
+        "-c", "--compression-format", default="gz", choices=["gz", "bz2", "xz"]
+    )
+    args = parser.parse_args()
+
+    if args.filenames is None and args.bootloader_slots is None:
+        raise Exception()
+
+    return args
+
+
+def match_varnames_to(filenames, tinfoil=None):
+    """
+    Match bitbake variable names to the given file names.
+
+    Parse the bitbake config data using the tinfoil API and find the
+    variable which is bound to the file name.
+
+    Return a dict with the variable names as keys and
+    file names as values.
+
+    :param filenames list: List of file names to look for.
+    :param tinfoil Tinfoil: bitbake tinfoil object.
+    """
+    return {
+        cvar.split("_FILENAME")[0]: fname
+        for cvar, fname in product(tinfoil.config_data, filenames)
+        if cvar.startswith("MBL_WKS")
+        and tinfoil.config_data.getVar(cvar) == fname
+    }
+
+
+def match_filenames_to(bootloader_slots, tinfoil=None):
+    """
+    Look up the bitbake variables corresponding to the bootloader slots.
+
+    Parse the bitbake config data using the tinfoil API and match the
+    MBL_WKS_BOOTLOADER variable to the given bootloader slot.
+
+    Return a dict with the variable name as the key and the file name as the
+    value.
+
+    :param bootloader_slots list: List of bootloader slots e.g [1, 2, fs]
+    :param tinfoil Tinfoil: bitbake tinfoil object.
+    """
+    output_map = {}
+    for conf_var, bl_slot in product(tinfoil.config_data, bootloader_slots):
+        if re.match(
+            r"MBL_WKS_BOOTLOADER.*{}_FILENAME".format(bl_slot), conf_var
+        ):
+            output_map[
+                conf_var.split("_FILENAME")[0]
+            ] = tinfoil.config_data.getVar(conf_var)
+    return output_map
+
+
+def copy_to_staging_dir(filename_var_map, staging_dir, tinfoil=None):
+    """
+    Copy files to a staging dir, renaming them as MBL_WKS_BOOTLOADER[n].
+
+    :param filename_var_map dict: Mapping of bitbake variable names to files.
+    :param staging dir Path: path to the dir where the files will be copied to.
+    """
+    deploy_dir_img = pathlib.Path(
+        tinfoil.config_data.getVar("DEPLOY_DIR_IMAGE")
+    )
+    for var, fname in filename_var_map.items():
+        src = deploy_dir_img / fname
+        dest = staging_dir / var
+        shutil.copy(str(src), str(dest))
+
+
+def create_payload_tar_archive(tar_archive_name, dir_path, compression_format):
+    """
+    Create the payload tar archive.
+
+    :param tar_archive_name str: Path to the tar archive.
+    :param dir_path Path: Path to the directory of the files to archive.
+    :param compression_format str: The compression format. Valid options are
+    gz, bz2 and xz.
```

Comment:
Like I wrote above, we don't want to compress the tar archive, but we probably do want to compress the files inside it... And like above, I'm not sure we should be offering so many compression formats... What does the update script support? For rootfs files it only supports `xz` I think.

Reply:
I'll remove the compression for the tar archive and compress the files with `xz`. I'll also remove the other options for now.